Assignment

Jerry-Hung opened the pull request.

Jerry-Hung committed:
Update README.md

Jerry-Hung committed:
Add files via upload

Jerry-Hung committed:
Update README.md

Jerry-Hung committed:
Update Program.java

Jerry-Hung committed:
Update CEvent.java

Jerry-Hung committed:
Update dateTime.java

Jerry-Hung committed:
Update Program.java

Jerry-Hung committed:
Add files via upload

Jerry-Hung committed:
Delete .classpath

Jerry-Hung committed:
Delete .project

Jerry-Hung committed:
Delete dateTime.java

Sir-Teo commented:
Let me answer your question first!
If I am a student using a digital calendar, I would definitely implement date and time, maybe todays weather(there are some digital calendars in my home that displays weather). I can add the lunar dates and fengshui in to the digital calendar. 
I don't think I understand your second question, asking me to describe time as an event. To my understanding, I will describe time as hours minutes and seconds, just like clocks do.

Comments:
1. I think you need a better README file to tell the users how to run your program on their computers. How does the user input their codes? Which file should the user run? Do they need to install some packages before running the program?

2. I really don't have previous experience in this domain and this isn't many related DSLs in this domain. One comment I can give to your grammar is that instead of describing time as 3:00 p.m. or 3:00 a.m., you can simply distinguish morning and afternoon by saying 3:00 and 15:00.

3. I think your project can be better if a GUI is implemented, so that the user can have a better sense at when will their events starts and ends.

Jerry-Hung committed:
Update Program.java

Jerry-Hung committed:
Update Program.java

Sir-Teo commented:
Your project looks good, let me answer your questions first!

If I am to describe a recurring event, ```Event A repeat [frequency]``` would definitely be a good option. Besides, I will probably say something like this:
```
Do Event A For [frequency]
```
As we discussed before, if you are looking for ways to describe a syntax, you can always consult domain experts. In this scenario, you can always ask people who use a calendar everyday and consult them for a more comprehensive feedback.

I hope this helps!

Zeng

Jerry-Hung committed:
Update Program.java

Jerry-Hung committed:
Delete CEvent.java

Jerry-Hung committed:
Delete Program.java

Jerry-Hung committed:
Add files via upload

Sir-Teo commented:
Hi Jerry,

Let me just answer your question! In your question, you said that the input is a bit weird for your project. And from your language design and overview, I found one of the possible input of your program,
```
English class: Starts at 9:00 a.m. Ends at 3:00 Beginning 8/1/2018
```
It looks pretty good overall, but I can give few suggestions():

1. Try to divide your input into sections and separate them with symbols. For example, this can turn into,
```
English class: Starts at 9:00 a.m. , Ends at 3:00 Beginning 8/1/2018
```
notice that the ```start``` part and the ```end```part is separated by comma.

2. Make information consistent. Notice that you put an ```a.m.``` right after ```9:00``` and did not put an ```a.m.``` or ```p.m.``` after ```3:00```.

3. Try make the sentence more fluent, think of what verbs can be use to describe these actions.

I hope this helps, good luck!

Jerry-Hung committed:
Update Program.java

Jerry-Hung committed:
Update Program.java

Added .ics file generator
Added reset function
Fixed repeating RRule bug
Added UID generator
/Checked with google calendar, works fine with google calendar.

Sir-Teo commented:
Hi Jerry,

It sounds like a good idea, Making a 4 hour repeat or a repeat for longer interval. Besides repeating, you can add more features to your program to make it more suitable for your intended users. 

Zeng

Jerry-Hung committed:
Update README.md

Jerry-Hung committed:
Update README.md

Jerry-Hung committed:
Create Sample Program 1

Jerry-Hung committed:
Create Sample Program 2

Jerry-Hung committed:
Update Program.java

Jerry-Hung committed:
Update README.md

Sir-Teo commented:
Hi Jerry,

I think lots of features can be implemented into an online calculator. For example, world time may be a good idea, by which I mean that the user will be able to tell the time at different places in the world. Another interesting feature I can think of is "count down". In this feature, the user will be able to set a date and then tell how many days left until that date.

Personally, I don't recommend an online calendar because it will require lot more work to do: you need to purchase a server as well as to obtain some knowledge of web development. More importantly, it is out of the scope of the course.

I hope this helps!

Zeng

Sir-Teo commented:
Hi Jerry,

Congratulations on finishing taking the SAT. Since you left no questions for me, then I will leave no critiques this time.

Best,
Zeng

Jerry-Hung committed:
Add files via upload
